lib/segment/src/index/field_index/histogram.rs

```diff
@@ -23,20 +23,34 @@ pub struct Counts {
     pub right: usize,
 }
 
-#[derive(PartialEq, PartialOrd, Debug, Clone, Serialize, Deserialize)]
+#[derive(PartialEq, Debug, Clone, Serialize, Deserialize)]
+#[repr(C)]
 pub struct Point<T> {
     pub val: T,
-    pub idx: usize,
+    pub idx: PointOffsetType,
 }
 
 impl<T: PartialEq> Eq for Point<T> {}
 
-#[allow(clippy::derive_ord_xor_partial_ord)]
-impl<T: PartialOrd + Copy> Ord for Point<T> {
-    fn cmp(&self, other: &Point<T>) -> Ordering {
-        (self.val, self.idx)
-            .partial_cmp(&(other.val, other.idx))
-            .unwrap()
+impl<T: PartialEq + PartialOrd> PartialOrd for Point<T> {
```

Comment:
Nit/maybe unrelated to this PR:
Maybe we should use `ordered_float` and mandate `Ord` and `Eq` for `T` instead of writing impls like these.

Reply:
It's a good idea and I like it. However, many changes are required because `OrderedFloat` will be a type of `NumericIndexInner` structure. I will think how to do it easier

Reply:
I cherry-picked this change into separate PR, where we can discuss it more
https://github.com/qdrant/qdrant/pull/4743